src/julia-parser.scm

```diff
@@ -1287,6 +1291,9 @@
                    (let* ((spec (with-space-sensitive (parse-subtype-spec s)))
                           (nb   (with-space-sensitive (parse-cond s))))
                      (begin0 (list 'bitstype nb spec)
+                             (let ((nt (peek-token s)))
+                               (if (or (newline? nt) (eqv? nt #\;))
+                                   (take-token s)))
                              (expect-end s "primitive type"))))))
```

Comment:
Since this exact code is repeated twice in this PR, I would define:
```scm
(define	(take-lineending s)
  (let ((nt (peek-token s)))
    (if (or (newline? nt) (eqv? nt #\;))
        (take-token s))
    s))
```
and then do e.g. `(expect-end (take-lineending s) "primitive type"))`